src/OpenSage.Game/Logic/Orders/OrderProcessor.cs

```diff
@@ -399,45 +408,50 @@ public void Process(IEnumerable<Order> orders)
 
-                        var container = gameObject.FindBehavior<OpenContainModule>();
-                        foreach (var unit in player.SelectedUnits)
+                        if (gameObject != null)
                         {
-                            if (!container.CanAddUnit(unit))
+                            var container = gameObject.FindBehavior<OpenContainModule>();
+                            foreach (var unit in player.SelectedUnits)
                             {
-                                continue; // this unit can't enter the container (kindof doesn't match, or there aren't enough slots)
+                                if (!container.CanAddUnit(unit))
+                                {
+                                    continue; // this unit can't enter the container (kindof doesn't match, or there aren't enough slots)
+                                }
+
+                                // TODO: Don't put it in container right now. Tell it to move towards container.
+                                //  use AIStateMachine EnterContainerState?
+                                //  deselect unit upon entry
+                                //unit.AIUpdate.SetTargetObject(gameObject);
+                                container.Add(unit);
                             }
-
-                            // TODO: Don't put it in container right now. Tell it to move towards container.
-                            //  use AIStateMachine EnterContainerState?
-                            //  deselect unit upon entry
-                            //unit.AIUpdate.SetTargetObject(gameObject);
-                            container.Add(unit);
                         }
-
                         break;
                     }
                 case OrderType.GatherDumpSupplies:
                     var supplyPointId = order.Arguments[0].Value.ObjectId;
                     var supplyPoint = _game.Scene3D.GameObjects.GetObjectById(supplyPointId);
 
-                    foreach (var unit in player.SelectedUnits)
+                    if (supplyPoint != null)
                     {
-                        var behavior = unit.FindBehavior<SupplyAIUpdate>();
-
-                        if (behavior is null)
+                        foreach (var unit in player.SelectedUnits)
                         {
-                            continue;
-                        }
+                            var behavior = unit.FindBehavior<SupplyAIUpdate>();
 
-                        if (supplyPoint.IsKindOf(ObjectKinds.SupplySource))
-                        {
-                            behavior.CurrentSupplySource = supplyPoint;
-                            behavior.SupplyGatherState = SupplyAIUpdate.SupplyGatherStates.SearchingForSupplySource;
-                        }
-                        else // if it's not a supply source, it's a supply center
-                        {
-                            behavior.CurrentSupplyTarget = supplyPoint;
```

Comment:
Would we not want to set this to null if for example the vehicles supply center was destroyed?

Reply:
Good point. I've had a go at improving it; although I'm not sure if I should really reset both. I imagine this will be fixed when we actually port this code.